src/main/kotlin/io/sc3/peripherals/posters/PosterItem.kt

```diff
@@ -0,0 +1,217 @@
+import io.sc3.peripherals.util.BaseNetworkSyncedItem
+import net.fabricmc.fabric.api.client.rendering.v1.TooltipComponentCallback
+import net.minecraft.client.MinecraftClient
+import net.minecraft.client.font.TextRenderer
+import net.minecraft.client.gui.DrawableHelper
+import net.minecraft.client.gui.tooltip.TooltipComponent
+import net.minecraft.client.item.TooltipContext
+import net.minecraft.client.item.TooltipData
+import net.minecraft.client.render.BufferBuilder
+import net.minecraft.client.render.GameRenderer
+import net.minecraft.client.render.Tessellator
+import net.minecraft.client.render.VertexConsumerProvider
+import net.minecraft.client.render.item.ItemRenderer
+import net.minecraft.client.util.math.MatrixStack
+import net.minecraft.entity.Entity
+import net.minecraft.entity.player.PlayerEntity
+import net.minecraft.item.ItemStack
+import net.minecraft.nbt.NbtCompound
+import net.minecraft.nbt.NbtElement
+import net.minecraft.nbt.NbtIo
+import net.minecraft.network.Packet
+import net.minecraft.text.Text
+import net.minecraft.util.Formatting
+import net.minecraft.world.World
+import java.io.ByteArrayOutputStream
+import java.io.DataOutputStream
+import java.security.MessageDigest
+import java.util.*
+
+class PosterItem(settings: Settings) : BaseNetworkSyncedItem("poster", settings) {
+
+  init {
+    TooltipComponentCallback.EVENT.register {
+      if (it is PosterTooltipData) {
+        PosterTooltipComponent(it.stack, MinecraftClient.getInstance().world)
+      } else {
+        null
+      }
+    }
+  }
+
+  data class PosterTooltipData(val stack: ItemStack): TooltipData
+
+  override fun getTooltipData(stack: ItemStack): Optional<TooltipData> = Optional.of(PosterTooltipData(stack))
+
+  override fun createSyncPacket(stack: ItemStack?, world: World?, player: PlayerEntity?): Packet<*>? {
+    val id = getPosterId(stack) ?: return null
+    val posterState = getPosterState(id, world)
+    return posterState?.getPlayerUpdatePacket(id, player)?.toS2CPacket()
+  }
+
+  override fun inventoryTick(stack: ItemStack, world: World, entity: Entity?, slot: Int, selected: Boolean) {
+    if (!world.isClient) {
+      val mapState = getOrCreatePosterState(stack, world)
+      if (mapState != null) {
+        if (entity is PlayerEntity) {
+          mapState.update(entity)
+        }
+      }
+    }
+  }
+
+  override fun getName(stack: ItemStack): Text = printData(stack)?.labelText ?: super.getName(stack)
+
+  override fun appendTooltip(stack: ItemStack, world: World?, tooltip: MutableList<Text>, context: TooltipContext) {
+    val id = getPosterId(stack)
+
+    val data = printData(stack) ?: return
+    data.tooltip?.let { tooltip.add(Text.literal(it)) }
+
+    if (context.isAdvanced) {
+      if (id != null) {
+        tooltip.add(Text.translatable("sc-peripherals.poster.id", id.take(8)).formatted(Formatting.GRAY))
+      } else {
+        tooltip.add(Text.translatable("sc-peripherals.poster.unknown").formatted(Formatting.GRAY))
```

Comment:
Can you change these to use `$modId` or `getTranslationKey()` instead? It will require changing the key to `item.sc-peripherals.poster.id` in the language file too (side note: just noticed we're not using datagen for that)

Assuming `translationKey` works for `getTranslationKey()` here:

```suggestion
        tooltip.add(Text.translatable("$translationKey.id", id.take(8)).formatted(Formatting.GRAY))
      } else {
        tooltip.add(Text.translatable("$translationKey.unknown").formatted(Formatting.GRAY))
```